datanode/entities/pagination.go

```diff
@@ -107,7 +107,7 @@ func NewCursorPagination(first *int32, after *string, last *int32, before *strin
 }
 
 func CursorPaginationFromProto(cp *v2.Pagination) (CursorPagination, error) {
-	if cp == nil {
+	if cp == nil || (cp != nil && cp.First == nil && cp.After == nil) {
```

Comment:
is that enough? shouldn't we check if all fields are nil instead?
```suggestion
	if cp == nil || (cp != nil && cp.First == nil && cp.After == nil && cp.Last == nil && cp.Before == nil) {
```

Reply:
The After and Before can be nil, the Last and After specify how many records to return, from the start and end. If After or Before are provided, then those cursors are used to determine where the position of the cursor in the data set.

Reply:
hum, alright, I suppose.